### PR TITLE
Added update tests for setOnInsert

### PR DESCRIPTION
**Pull request by vic-tsang**
This PR contains: 
- update tests for $setOnInsert

Ref:
- issue #399

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_command_contexts.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        expected={"value": {"_id": 1, "x": 100}},
+        msg="Should return new doc with $setOnInsert fields",
+    ),
+    UpdateTestCase(
+        id="upsert_return_old",
+        query={"_id": 2},
+        update={"$setOnInsert": {"x": 100}},
+        upsert=True,
+        expected={"value": None},
+        msg="Should return null when no doc existed before",
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_command_contexts.py]: there is no setup doc in this and even if the both runs together _id 2 will not exist and should insert the data

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        msg="Should preserve Decimal128 -0",
+    ),
+    UpdateTestCase(
+        id="null_vs_missing",
+        query={"_id": 1},
+        update={"$setOnInsert": {"field": None}},
+        upsert=True,
+        expected=[{"_id": 1, "field": None}],
+        msg="Should store null value (field present, not missing)",
+    ),
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py:63]: add double-NaN case

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_errors.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+"""
+Tests for $setOnInsert update operator - errors.
+
+Covers conflicting paths, conflict with $set, _id immutability, and conflicting
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_errors.py:4]: there is no set conflict cases

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_field_paths.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        # not array indices. Arrays are only created when updating an existing array.
+        expected=[{"_id": 1, "arr": {"0": "value"}}],
+        msg="Should create object with numeric key on insert",
+    ),
+    UpdateTestCase(
+        id="dollar_prefix_field",
+        query={"_id": 1},
+        # Tests that "$invalid" is treated as a field name inside $setOnInsert,
+        # not as an unknown update operator.
+        update={"$setOnInsert": {"$invalid": 1}},
```
Comment (manojnks) [inline documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_field_paths.py:40]: $set fails this case and it works here. just wanna make sure it correct behavior

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py (hunk 2 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py:63]: added

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_errors.py (hunk 3 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_errors.py:4]: added

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_command_contexts.py (hunk 1 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_command_contexts.py]: testing findAndModify not update. Removed

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_field_paths.py (hunk 4 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_field_paths.py:40]: confirmed. $set is more strict in terms of what is allowed.

**Review by manojnks — APPROVED** [review]

File: documentdb_tests/compatibility/tests/core/operator/update/fields/test_fields_cross_operator_conflicts.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
+"""
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/fields/test_fields_cross_operator_conflicts.py]: [Minor] Three open PRs (#566, #570, #576) each create this same file with different contents. Whichever merges first will land its version; the other two will hit merge conflicts and need rebase. Consider coordinating with the authors of #566 and #570 to land one canonical file.

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    UpdateTestCase(
+        id="preserves_decimal128_neg_zero",
+        query={"_id": 1},
+        update={"$setOnInsert": {"field": DECIMAL128_NEGATIVE_ZERO}},
+        upsert=True,
+        expected=[{"_id": 1, "field": DECIMAL128_NEGATIVE_ZERO}],
+        msg="Should preserve Decimal128 -0",
+    ),
+    UpdateTestCase(
+        id="null_vs_missing",
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py]: [Minor] The case ID and message claim a "null vs missing" distinction, but the assertion only verifies `field: None` is present — it doesn't actually exercise the missing path. Either rename to `preserves_null` (drop the "vs missing" framing), or add a sibling test using `{$exists: True}` filter to prove the field is stored as null and not missing.

File: documentdb_tests/compatibility/tests/core/operator/update/fields/test_fields_cross_operator_conflicts.py (hunk 5 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/update/fields/test_fields_cross_operator_conflicts.py]: merged file contents into https://github.com/documentdb/functional-tests/pull/566 removing this file from this PR.

File: documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py (hunk 6 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/update/fields/setOnInsert/test_setOnInsert_data_types.py]: renamed to preserves_null

**Review by eerxuan — APPROVED** [review]
